db/sql.php

```diff
@@ -122,12 +120,7 @@ function type($val) {
 	function value($type,$val) {
 		switch ($type) {
 			case self::PARAM_FLOAT:
-				if (is_string($val)) {
-					$val=str_replace(',','.',preg_replace('/([.,])(?!\d+$)/','',$val));
-					if (strlen($val)-(int)(bool)strpos($val,'.') <= $this->precision)
-						$val=(float) $val;
-				}
-				return $val;
+				return rtrim(sprintf('%F',$val),'0');
```

Comment:
That line returns `n.` instead of `n` when `$val = n` and `n` is an integer. Is this the expected behaviour?

`$val` | Result
-|-
`1` | `1.`
`1.0` | `1.`
`1.10` | `1.1`